src/blr/cli/tagfastq.py

```diff
@@ -88,6 +89,22 @@ def main(args):
     logger.info("Finished")
 
 
+def search_bc(iterator: iter, header: str, cache: dict, maxiter: int = 10):
+    # Check it header is stored in cache. If not move forward on step in iterator at look again.
+    iteration = 0
+    while header not in cache and iteration < maxiter:
```

Comment:
Note that every uncorrected barcode that exists in the input barcodes, but is never searched for, will remain in the cache. Perhaps this doesn’t matter that much, but you could check whether the value you get from `next(iterator)` is the one you are searching for and then clear the cache because you know that you’ll not need the earlier barcodes anymore.

Reply:
Implemented in new PR.